code/game/machinery/computer/sm_monitor.dm

```diff
@@ -0,0 +1,142 @@
+/obj/machinery/computer/sm_monitor
+	name = "supermatter monitoring console"
+	desc = "Used to monitor supermatter shards."
+	icon_keyboard = "power_key"
+	icon_screen = "smmon_0"
+	circuit = /obj/item/circuitboard/sm_monitor
+	light_color = LIGHT_COLOR_YELLOW
+	/// Cache-list of all supermatter shards
+	var/list/supermatters = list()
+	/// Last status of the active supermatter for caching purposes
+	var/last_status
+	/// Reference to the active shard
+	var/obj/machinery/power/supermatter_shard/active
+
+/obj/machinery/computer/sm_monitor/Destroy()
+	active = null
+	return ..()
+
+/obj/machinery/computer/sm_monitor/attack_ai(mob/user)
+	attack_hand(user)
+
+/obj/machinery/computer/sm_monitor/attack_hand(mob/user)
+	add_fingerprint(user)
```

Comment:
I feel like this shouldn't silently add a fingerprint if the computer's broken or without power. The player doesn't get any feedback if that's the case

Reply:
Youre still touching the machine, and every instance of `attack_hand` should be fingerprinted, not just for the detective, but admins too